tools/revpi-set-mac

```diff
@@ -36,22 +36,32 @@ case $dev_type in
         size=256
         eeprom_file='/usr/share/revpi/lan9514.bin'
         ;;
+    ks8851)
+        magic='0x8851'
+        # offset needs to be 7 to 2 (first to last byte)
+        offset_delta=-8
+        offset_multiplier=-1
+        ;;
     *)
         1>&2 echo "Unsupported device type: '$dev_type'"
         exit 1
         ;;
 esac
 
 # write golden image to EEPROM
-if [ -f "${eeprom_file}" ]; then
-    ethtool -E "${iface}" magic "${magic}" offset 0 length "${size}" < "${eeprom_file}"
-else
-    echo 1>&2 "ERROR: Golden image not found: '${eeprom_file}"
-    exit 1
+if [ -n "${eeprom_file:-}" ]; then
+    if [ -f "${eeprom_file}" ]; then
```

Comment:
Any reason not to combine the two if-clauses into one, i.e. `if [ -n "${eeprom_file:-}" ] && [ -f "${eeprom_file}" ] ; then`?
It would avoid the indentation change below.

Reply:
Yes. The else part should be only taken into account if there was a golden image specified (eg. there is none for ks8851)